src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -3708,267 +3443,146 @@ void PredictSystemLoads(EnergyPlusData &state,
-                    for (LoopNode = 1; LoopNode <= RoomAirflowNetworkZoneInfo(ZoneNum).NumOfAirNodes; ++LoopNode) {
-                        RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).AirTempT1 = RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).AirTemp;
-                        RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).HumRatW1 = RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).HumRat;
-                    }
+                if (thisTempControlledZone.HeatOffFlag && state.dataZoneEnergyDemand->ZoneSysEnergyDemand(ZoneNum).TotalOutputRequired <= 0.0) {
+                    thisTempControlledZone.HeatModeLast = true;
+                } else {
+                    thisTempControlledZone.HeatModeLast = false;
                 }
             }
-            TempDepZnLd(ZoneNum) = TempDepCoef;
-            TempIndZnLd(ZoneNum) = TempIndCoef;
         }
+    }
+}
+void ZoneSpaceHeatBalanceData::predictSystemLoad(
+    EnergyPlusData &state,
+    bool const shortenTimeStepSys,
+    bool const useZoneTimeStepHistory, // if true then use zone timestep history, if false use system time step
+    Real64 const priorTimeStep,        // the old value for timestep length is passed for possible use in interpolating
+    int zoneNum,
+    int spaceNum)
+{
+    assert(zoneNum > 0);
+    this->updateTemperatures(state, shortenTimeStepSys, useZoneTimeStepHistory, priorTimeStep, zoneNum, spaceNum);
+
+    Real64 volume = 0.0;
+    if (spaceNum > 0) {
+        volume = state.dataHeatBal->space(spaceNum).Volume;
+    } else {
+        volume = state.dataHeatBal->Zone(zoneNum).Volume;
+    }
+    this->AirPowerCap = volume * state.dataHeatBal->Zone(zoneNum).ZoneVolCapMultpSens *
+                        Psychrometrics::PsyRhoAirFnPbTdbW(state, state.dataEnvrn->OutBaroPress, this->MAT, this->ZoneAirHumRat) *
+                        Psychrometrics::PsyCpAirFnW(this->ZoneAirHumRat) / (state.dataHVACGlobal->TimeStepSys * DataGlobalConstants::SecInHour);
+    Real64 RAFNFrac = 0.0;
 
-        // Calculate the predicted zone load to be provided by the system with the given desired zone air temperature
-        CalcPredictedSystemLoad(state, ZoneNum, RAFNFrac);
+    // Calculate the various heat balance sums
 
-        // Calculate the predicted zone load to be provided by the system with the given desired humidity ratio
-        CalcPredictedHumidityRatio(state, ZoneNum, RAFNFrac);
+    // NOTE: SumSysMCp and SumSysMCpT are not used in the predict step
+    this->calcZoneOrSpaceSums(state, false, zoneNum, spaceNum);
+
+    // Sum all convective internal gains except for people: SumIntGainExceptPeople
+    if (spaceNum == 0 && state.dataHybridModel->FlagHybridModel_PC) {
+        this->SumIntGainExceptPeople = 0.0;
+        this->SumIntGainExceptPeople = InternalHeatGains::SumAllInternalConvectionGainsExceptPeople(state, zoneNum);
     }
```

Comment:
For now (and maybe always) special features like the hybrid zone model and non-mixed room air models are evaluated only for the zone heat balance.